tests/optimization/bfgs.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+   Copyright (c) 2009-2016, Ryan H. Lewis
+   All rights reserved.
+
+   This file is part of Elemental and is under the BSD 2-Clause License, 
+   which can be found in the LICENSE file in the root directory, or at 
+   http://opensource.org/licenses/BSD-2-Clause
+*/
+#include <utility>
+#include <El.hpp>
+#include <El/optimization/bfgs.hpp>
+#define CATCH_CONFIG_RUNNER
+#include <catch.hpp>
+
+using namespace El;
+
+template< typename T>
+std::pair< DistMatrix<T>, T>
+LogisticRegression( const DistMatrix<T>& X, const DistMatrix<T>& y, const T lambda=1){
+  const std::function< T(const DistMatrix<T>&)>
+  logistic_function = [&](const DistMatrix<T>& theta){
+     auto nrm = El::Norm(theta);
+     El::DistMatrix<T> y(theta);
+     El::DistMatrix<T> r(X.Height(), 1);
+     El::Gemv(El::NORMAL, T(-1), X, theta, T(0), r);
+     for( auto i = 0; i < y.Height(); ++i){ r.Set(i, 0, y.Get(i,0));  }
+
+     T v = lambda*nrm*nrm;
+     for(auto i = 0; i < y.Height(); ++i){ v += El::Log(T(1) + El::Exp(-1*y.Get(i,0)*r.Get(i,0))); }
```

Comment:
I'm going to temporarily remove this and bring it back later. I have changed this on a separate branch. It's still not working perfectly, and I'm not confident I have the right formulae. I'm interested in just appreciating that the core BFGS procedure is satisfactory.